packages/stylelint-plugin/src/util/colors.ts

```diff
@@ -0,0 +1,19 @@
+import tinycolor from "tinycolor2"
+
+/**
+ * Normalises a color string to a common representation.
+ * Use this for making comparisons between color strings.
+ * For example: `normaliseColor("#fff") === normaliseColor("#ffffff")` will be true.
+ *
+ * @param color a string representation of a color, which can be in any kind of format like `#fff`, `blue`, `rgba(200, 60, 60, 0.3)`, `hsl(195, 100%, 50%)` etc. See https://www.npmjs.com/package/tinycolor2.
+ * @returns a string representation of the color (at the time of writing, it is a hex or rgba string), or null if the string you passed in wasn't a color.
+ */
+export function normaliseColor(color: string): string | null {
+  const parsedColor = tinycolor(color)
+  if (!parsedColor.isValid()) return null
+
+  // If alpha is less than 1, we should stringify as rgba so that the alpha channel isn't lost.
+  // Otherwise we should use hex.
```

Comment:
@jasononeil yeah i went down that rabbit hole and tried many libraries lol!
There we're pros and cons to each (package size, support for multiple color functions like HSL), but one of the things i wanted was a return value that could also be used within CSS because i think this does happen, and 4 channel hex values aren't supported.